enhancements/sig-architecture/64-placementStrategy/README.md

```diff
@@ -0,0 +1,220 @@
+
+### Goals
+
+1. Add a new field placementStrategy to the placement APIs to let end uer define the roll out strategy for the selected clusters.
+
+### Non-Goals
+
+1. The new Placement APIs field placementStrategy has no control over applying the workloads.
+1. The new Placement APIs field placementStrategy has no control over scheduling the workloads.
+
+## Proposal
+
+### User Stories
+
+#### Story 1: As end user I want to define a subset of the selected clusters as a canary clusters.
+#### Story 2: As end user I want to divide the selected clusters to groups then a workloads applier APIs can use those cluster groups to roll out the workloads. 
+
+### Placement placementStrategy API
+
+Adding PlacementStrategy to PlacementSpec. PlacementStrategy has a labelSelector to identify canary clusters. CanaryClusters must be a subset of the selected clusters based on predicates and ClusterSets. MaxConcurrency is a percentage of the total selected clusters that will divide the clusters to groups(batches). Example; for a total 100 clusters selected, MaxConcurrency equal to 20% will divide the selected clusters into 5 groups each group consist of 20 clusters.
+```go
+type PlacementSpec struct {
+	// ClusterSets represent the ManagedClusterSets from which the ManagedClusters are selected.
+	// +optional
+	ClusterSets []string `json:"clusterSets,omitempty"`
+
+	// Predicates represent a slice of predicates to select ManagedClusters. The predicates are ORed.
+	// +optional
+	Predicates []ClusterPredicate `json:"predicates,omitempty"`
+            ...
+ 
+	// PlacementStrategy define canary clusters as subset of the selected clusters and divide the selected clusters to groups.
+	// +optional
+	PlacementStrategy PlacementStrategy `json:"placementStrategy,omitempty"`
+}
+
+type PlacementStrategy struct {
+	// Label selector for canary clusters as subset of the selected clusters from predicates and ClusterSets.
+	// +optional
+	CanaryClusters metav1.LabelSelector `json:"canaryClusters,omitempty"`
```

Comment:
Not sure if  the name is confusing, could be something like **FirstGroupSelector** ? but the idea is those selected clusters need to be in the first cluster group.  
"Optionally, we we use another placement and define that placement is a affinity" I have 2 thoughts on that;
1- UX; as a user why do I need to define 2 placements ? usually canary is a small subset of the large cluster set that I want to select. If as user will define 2 placements as example; **1-** first placement should **prod** and **Canary**. **2-** second placement should **prod** and **Not Canary**. I feel its more pushing effort to end user while we can easily handle that. 
2- What if the affinity placement has affinity placement and so on ? it will add complexity to select all clusters

Reply:
For the canary use case, the cluster set is not only a subset of the larger set, but also excluded from the rest of the larger set in terms of grouping. Placement affinity probably does the sub selection part, but doesn't provide grouping isolation?

Reply:
How about name it affinityClusterGroup? It gives the meaning of dependent cluster group

Reply:
I think it looks like a `decisionGroup` which is just to sub-divide the decisions to multiple groups. But I am still not fully convinced that the pros of having a `decisionGroup` than using another placement to generate another set of decision.

Reply:
the canary update which I understand is similar to [OCP worker nodes rollout canary update](https://docs.openshift.com/container-platform/4.12/updating/update-using-custom-machine-config-pools.html). 
End user selects a set of clusters as canary cluster to update firstly, and evaluates the workloads health after the update is finished, and then to determine whether to update another selected clusters. In most cases the canary clusters are selected rather than a group of random clusters.
During the updating of the canary clusters, end user may not expect they are upgraded at the same time if the count of selected cluster is large, and wants they are upgrade under a maximum concurrently updating number. I prefer to call this update as rollout. In rollout update, the group of clusters can be randomly selected.

Reply:
yes agree, the canary has been removed and affinity cluster selection is added. the affinity cluster selection has its own label selector.

Reply:
I'm trying to think about about how this gets implemented... The controller that is consuming the PlacementDecisions would need to understand index 0 is possibly a canary(affinity Group), but I'm thinking that how the groups are consumed is dependent on the controller that is consuming decisions, and Canary or AffinityGroup is an implementation detail that can be used in different ways depending on what the controller is using placement to do.

For example:
1. The implementation requirement might be that it processes ONE cluster in each group before it starts to process all the others. So the ManifestWorkSet controller, will try the first cluster in each group before it tries the rest.
2. The implementation requirement might be that 4x clusters in the first group (this may change over time as order is not guaranteed) should be upgraded and if successful, continue to the rest.
3. The implementation requirement might be to use a label to find the canary (first) clusters, the controller may use a separate placement (borrowing from @qiujian16 's suggestion) and will remove these clusters when it starts to process the groups.

So  we leave it up to the implementation to decided how they want to process clusters, since it will be different for different scenarios and controllers.  We can provide some examples like the three above of how you might use the grouping to accomplish different canary scenarios.

Reply:
Okay then, it make more sense to remove the ClusterGroupType as it is up to the controller to decide how to roll out the workload.  I will just add a label for the affinity groups to distinguish between the cluster groups . AND will add the examples mentioned above to the "Workload applier" section.